apstools/utils.py

```diff
@@ -727,6 +729,65 @@ def split_quoted_line(line):
     return parts
 
 
+def summarize_runs(since=None, db=None):
+    """
+    Report bluesky run metrics from the databroker.
+
+    * How many different plans?
+    * How many runs?
+    * How many times each run was used?
+    * How frequently?  (TODO:)
+
+    PARAMETERS
+
+    since (str) :
+        Report all runs since this ISO8601 date & time (default: ``1995``)
+    db (object) :
+        Instance of ``databroker.Broker()``
+        (default: ``db`` from the IPython shell)
+    """
+    db = db or ipython_shell_namespace()["db"]
+    since = since or "1995"
+    cat = db.v2.search(databroker.queries.TimeRange(since=since))
+    plans = defaultdict(list)
+    t0 = time.time()
+    for n, uid in enumerate(cat):
+        t1 = time.time()
+        run = cat[uid]      # this step is very slow (0.01 - 0.5 seconds each!)
+        t2 = time.time()
+        plan_name = run.metadata["start"].get("plan_name", "unknown")
+        dt = datetime.datetime.fromtimestamp(run.metadata["start"]["time"]).isoformat()
+        scan_id = run.metadata["start"].get("scan_id", "unknown")
+        plans[plan_name].append(
+            dict(
+                plan_name=plan_name,
+                dt=dt,
+                time_start=dt,
+                uid=uid,
+                scan_id=scan_id,
+            )
+        )
+        logger.debug(
```

Comment:
Codacy found an issue: [Too many arguments for logging format string](https://app.codacy.com/gh/BCDA-APS/apstools/pullRequest?prid=5832153)